pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1346,14 +1346,9 @@ func (c *PodAffinityChecker) satisfiesExistingPodsAntiAffinity(pod *v1.Pod, meta
 	if predicateMeta, ok := meta.(*predicateMetadata); ok {
 		topologyMaps = predicateMeta.topologyPairsAntiAffinityPodsMap
 	} else {
-		// Filter out pods whose nodeName is equal to nodeInfo.node.Name, but are not
-		// present in nodeInfo. Pods on other nodes pass the filter.
-		filteredPods, err := c.podLister.FilteredList(nodeInfo.Filter, labels.Everything())
-		if err != nil {
-			errMessage := fmt.Sprintf("Failed to get all pods: %v", err)
-			klog.Error(errMessage)
-			return ErrExistingPodsAntiAffinityRulesNotMatch, errors.New(errMessage)
-		}
+		filteredPods := nodeInfo.Pods()
```

Comment:
We shouldn't do that because the existing pod without any affinity may be anti-affinity with the pod we are scheduling.

Reply:
> We shouldn't do that because the existing pod without any affinity may be anti-affinity with the pod we are scheduling.

nodeInfo.PodsWithAffinity() include the pods that have podAffinity or podAntiAffinity.

Reply:
Sry, missing read~

Reply:
> So here can just get the affinity pods on the node.
> filteredPods := nodeInfo.PodsWithAffinity()

> filteredPods := nodeInfo.Pods() can be applied to satisfiesPodsAffinityAntiAffinity func,

Sorry, I'm confused, what are you suggesting?

Reply:
> > So here can just get the affinity pods on the node.
> > filteredPods := nodeInfo.PodsWithAffinity()
> 
> > filteredPods := nodeInfo.Pods() can be applied to satisfiesPodsAffinityAntiAffinity func,
> 
> Sorry, I'm confused, what are you suggesting?

For satisfiesExistingPodsAntiAffinity func, it just check whether the incoming pod schedule onto this node would break any anti-affinity terms by the existing pod on the node. It will use the incoming pod's namespaces and label to match the existing pod's anti-affinity term on the node. 
And the nodeInfo already has all pods with affinity. So we can use it directly.
```
filteredPods := nodeInfo.PodsWithAffinity()
```

And in satisfiesPodsAffinityAntiAffinity func, it will check the incoming pod's affinity/anti-affinity terms whether match the pod's label and namespaces. The nodeInfo did not have pods with namespaces.

Reply:
> > So here can just get the affinity pods on the node.
> > filteredPods := nodeInfo.PodsWithAffinity()
> 
> > filteredPods := nodeInfo.Pods() can be applied to satisfiesPodsAffinityAntiAffinity func,
> 
> Sorry, I'm confused, what are you suggesting?

The suggestion maybe need to consider the assumed pods.